app/alignment/utils.py

```diff
@@ -413,3 +415,105 @@ def log_solution_choice(solution_choice_uuid, shared_solutions):
         raise DatabaseError(
             message="An error occurred while saving user b's solution choice to the db."
         )
+
+
+def build_shared_impact_details_response(impact_iri):
+    """
+    Get the detals for a shared impact.
+
+    Includes checks that the IRI is found and of the correct type, e.g. not a solution IRI.
+
+    Parameters
+    ==========
+    impact_iri - the IRI for the shared impact
+
+    Returns
+    ==========
+    JSON:
+    - the effect title
+    - the image url (if available)
+    - the long description for the effect
+    - the sources for the effect
+    - a list of personal values that are related to the effect
+    """
+
+    G = current_app.config["G"].copy()
+    nx = network_x_utils()
+    impact = None
+
+    for node in G.nodes:
+        if G.nodes[node]["iri"][24:] == impact_iri:
+            nx.set_current_node(G.nodes[node])
+            if "effect" in G.nodes[node]["all classes"]:
+                impact = {
+                    "effectTitle": G.nodes[node]["label"],
+                    "imageUrl": nx.get_image_url_or_none(),
+                    "longDescription": nx.get_description(),
+                    "effectSources": nx.get_causal_sources(),
+                    "relatedPersonalValues": map_associated_personal_values(
+                        G.nodes[node]["personal_values_10"]
+                    ),
+                }
+                break
+            else:
+                raise InvalidUsageError(message="IRI does not refer to an impact.")
+
+    if not impact:
+        raise NotInDatabaseError(message="The shared impact IRI cannot be found.")
+
+    return impact
+
+
+def build_shared_solution_details_response(solution_iri):
+    """
+    Gets the details for a shared solution.
+
+    Includes checks that the IRI is found and of the correct type, e.g. not an impact IRI.
+
+    Parameters
+    ==========
+    solution_iri - the IRI for the shared solution
+
+    Returns
+    ==========
+    JSON:
+    - the solution title
+    - the image url (if available)
+    - the long description for the solution
+    - the sources for the solution
+    - the solution type
+    """
+
+    G = current_app.config["G"].copy()
+    nx = network_x_utils()
+    solution = None
+
+    for node in G.nodes:
+        if G.nodes[node]["iri"][24:] == solution_iri:
```

Comment:
this should use this function instead of hard coding '24:' to get the iri ID https://github.com/ClimateMind/climatemind-backend/blob/0299b7106f42bdb06942a4e7485143c63fb893ce/app/network_x_tools/network_x_utils.py#L23